client/src/resources/amount.rs

```diff
@@ -197,3 +198,86 @@ mod deserialize_amount_tests {
         assert!(amount.is_err());
     }
 }
+
+/// Recreating ParseIntError since its interior is private
+/// https://doc.rust-lang.org/src/core/num/mod.rs.html#3949
+#[derive(Debug)]
+pub enum ParseIntError {
+    Empty,
+    InvalidDigit,
+    Overflow,
+    Underflow,
+}
+
+// There is a built in __description() method that gives
+// the error as a string, however that is only availible on
+// nightly...
+// So I went with @amosbatto's hack here:
+// https://github.com/rust-lang/rust/issues/22639
+impl From<num::ParseIntError> for ParseIntError {
+    fn from(error: num::ParseIntError) -> Self {
+        match error.to_string().as_ref() {
+            "cannot parse integer from empty string" => ParseIntError::Empty,
+            "invalid digit found in string" => ParseIntError::InvalidDigit,
+            "number too large to fit in target type" => ParseIntError::Overflow,
+            "number too small to fit in target type" => ParseIntError::Underflow,
+            _ => panic!("Unexpected error variant!"),
+        }
+    }
+}
+
+impl FromStr for Amount {
+    type Err = ParseIntError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.rfind('.') {
```

Comment:
This is copy-pasta from the deserialize impl, I think. It would be better to switch to using this implementation to implement the Deserialize function and map to the correct errors that it expects there.

Reply:
Yes, I admit I copy-pasta this. To clarify, are you saying I should keep FromStr and then utilize the FromStr implementation as part of deserialize, removing the code duplication, then mapping the errors of FromStr to appropriately match the Deserialize error type?